pkg/geo/geomfn/validity_check.go

```diff
@@ -64,6 +66,30 @@ func IsValidDetail(g geo.Geometry, flags int) (ValidDetail, error) {
 	}, nil
 }
 
+// IsValidTrajectory returns whether a geometry encodes a valid trajectory
+func IsValidTrajectory(line geo.Geometry) (bool, error) {
+	t, err := line.AsGeomT()
+	if err != nil {
+		return false, err
+	}
+	lineString, ok := t.(*geom.LineString)
+	if !ok {
+		return false, errors.AssertionFailedf("expected linestring, got %T", t)
+	}
+	mIndex := t.Layout().MIndex()
+	if mIndex < 0 {
+		return false, errors.New("linestring doesn't have measures (M coordinates)")
```

Comment:
nit: we want full sentences / no contractions here, let's do "LineString does not have M coordinates"